tests/e2e/performance/test_single_pvc_clone_perf.py

```diff
@@ -0,0 +1,175 @@
+Test to verify clone creation and deletion performance
+"""
+import logging
+import pytest
+
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    skipif_ocs_version
+)
+from ocs_ci.framework.testlib import (
+    performance, E2ETest
+)
+from ocs_ci.ocs.resources import pvc
+from ocs_ci.ocs.resources import pod
+from tests import helpers
+from ocs_ci.framework.pytest_customization.marks import (
+    bugzilla
+)
+
+logger = logging.getLogger(__name__)
+
+
+@skipif_ocs_version('<4.6')
+@pytest.mark.parametrize(
+    argnames=["interface_type", "pvc_size", "file_size"],
+    argvalues=[
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, '1', '600M'], marks=pytest.mark.polarion_id('OCS-2340')
+        ),
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, '25', '15GB'], marks=pytest.mark.polarion_id('OCS-2340')
+        ),
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, '50', '30GB'], marks=pytest.mark.polarion_id('OCS-2340')
+        ),
+        pytest.param(
+            *[constants.CEPHBLOCKPOOL, '100', '60GB'], marks=pytest.mark.polarion_id('OCS-2340')
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, '1', '600M'], marks=pytest.mark.polarion_id('2341')
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, '25', '15GB'], marks=pytest.mark.polarion_id('2341')
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, '50', '30GB'], marks=pytest.mark.polarion_id('2341')
+        ),
+        pytest.param(
+            *[constants.CEPHFILESYSTEM, '100', '60GB'], marks=pytest.mark.polarion_id('2341')
+        )
+    ]
+)
+@performance
+@bugzilla('1882504')
+class TestPVCSingleClonePerformance(E2ETest):
+    """
+    Test to verify single PVC clone creation and deletion performance
+    """
+
+    @pytest.fixture(autouse=True)
+    def base_setup(self, interface_type, pvc_size, pvc_factory, pod_factory):
+        """
+        create resources for the test
+        Args:
+            interface_type(str): The type of the interface
+                (e.g. CephBlockPool, CephFileSystem)
+            pvc_size: Size of the created PVC
+            pvc_factory: A fixture to create new pvc
+            pod_factory: A fixture to create new pod
+        """
+        self.pvc_obj = pvc_factory(
+            interface=interface_type,
+            size=pvc_size,
+            status=constants.STATUS_BOUND
+        )
+        self.pod_obj = pod_factory(
+            interface=interface_type,
+            pvc=self.pvc_obj,
+            status=constants.STATUS_RUNNING
+        )
+
+    def test_clone_create_delete_performance(self, interface_type, pvc_size, file_size, teardown_factory):
+        """
+        Create single clone for an existing pvc,
+        Measure clone creation times
+        Delete the created clone
+        Measure clone deletion time
+        Note: by increasing max_num_of_clones value you increase created clones number
```

Comment:
I would rather put `max_num_of_clones` as an arg to this function for better reusability. So the callers can adjust the arg based on the demand

Reply:
@ypersky1980 Not sure I understand where max_num_of_clones is configurable